.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,38 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ main ]
```

Comment:
is this the correct syntax?

Reply:
I think I pulled this from the sample config they gave me so I think so?!